data-weave-lang-server/src/main/scala/org/mule/weave/lsp/ui/wizard/QuickPickWidgetBuilder.scala

```diff
@@ -0,0 +1,87 @@
+package org.mule.weave.lsp.ui.wizard
+
+import org.mule.weave.lsp.extension.client.WeaveButton
+import org.mule.weave.lsp.extension.client.WeaveLanguageClient
+import org.mule.weave.lsp.extension.client.WeaveQuickPickItem
+import org.mule.weave.lsp.extension.client.WeaveQuickPickParams
+import org.mule.weave.lsp.ui.utils.Buttons.Button
+
+import scala.collection.JavaConverters.asScalaBufferConverter
+import scala.collection.JavaConverters.seqAsJavaListConverter
+
+class QuickPickWidgetBuilder[A](val client: WeaveLanguageClient) extends WidgetBuilder[List[String], A] {
+
+
+  var totalSteps: Int = _
+
+  var stepNumber: Int = _
+
+  var title: String = _
+
+  var buttons: Seq[(Button, (A => WidgetResult[A]))] = Seq()
+
+  var items: Seq[(WeaveQuickPickItem, (A => WidgetResult[A]))] = Seq()
+
+  var resultMapping: (A,List[String]) => A = _
+
+  override def title(title: String): QuickPickWidgetBuilder[A] = {
+    this.title = title
+    this
+  }
+
+  override def stepNumber(number: Int): QuickPickWidgetBuilder[A] = {
+    this.stepNumber = number
+    this
+  }
+
+  override def totalSteps(number: Int): QuickPickWidgetBuilder[A] = {
+    this.totalSteps = number
+    this
+  }
+
+  override def button(button: Button, function: A => WidgetResult[A]): QuickPickWidgetBuilder[A] = {
+    buttons = buttons :+ (button, function)
+    this
+  }
+
+  def item(item: WeaveQuickPickItem, function: A => WidgetResult[A]): QuickPickWidgetBuilder[A] = {
+    this.items = items :+ (item,function)
+    this
+  }
+
+  override def create(): Widget[A] = {
+    new QuickPickWidget(client, this)
+  }
+
+  override def result(function: (A,List[String]) => A): QuickPickWidgetBuilder[A] = {
+    resultMapping = function
+    this
+  }
+}
+
+class QuickPickWidget[T](languageClient: WeaveLanguageClient, quickInputWidgetBuilder: QuickPickWidgetBuilder[T]) extends Widget[T] {
+
+  var buttons: Map[String, (T => WidgetResult[T])] = Map()
+  var items: Map[String, (T => WidgetResult[T])] = Map()
+
+  override def show(resultToFill: T): WidgetResult[T] = {
```

Comment:
Need to refactor these, a lot of common stuff between this one and the InputWidget